src/feedspora/generic_client.py

```diff
@@ -182,75 +151,77 @@ def shorten_url(self, the_url):
                 if isinstance(exception, AssertionError):
                     all_shorteners = ' '.join(shortener.available_shorteners)
                     logging.error('URL shortener %s is unimplemented!',
-                                  self._url_shortener)
+                                  self._account['url_shortener'])
                     logging.info('Available URL shorteners: %s',
                                  all_shorteners)
                 else:
-                    logging.error('Cannot shorten URL %s with %s: %s', the_url,
-                                  self._url_shortener, str(exception))
+                    logging.error('Cannot shorten URL %s with %s: %s',
+                                  the_url, self._account['url_shortener'],
+                                  str(exception))
                 to_return = the_url
             # pylint: enable=broad-except
 
         return to_return
 
-    # pylint: disable=too-many-branches
+    def set_option_defaults(self, account):
+        '''
+        Set default values for options common to all clients
+        :param account:
+        '''
+
+        if 'max_posts' not in account:
```

Comment:
For the sake of clarity, these default could be stored in a dict, e.g.:

```python
defaults = { 'max_post': 0, 'max_tags': 100, 'post_prefix': '' ....}
for key, default_value in defaults.items():
  if key not in account:
    account[key] = default_value
```

Reply:
Nice.  I'll change the implementation to do it that way.

Reply:
Done